src/vs/workbench/contrib/notebook/browser/notebookServiceImpl.ts

```diff
@@ -165,12 +163,12 @@ export class NotebookProviderInfoStore extends Disposable {
 				const notebookOptions: INotebookEditorOptions = { ...options, cellOptions };
 				return { editor: NotebookEditorInput.create(this._instantiationService, notebookUri, notebookProviderInfo.id), options: notebookOptions };
 			};
-			const notebookEditorDiffFactory: DiffEditorInputFactoryFunction = (diffEditorInput: DiffEditorInput, options, group) => {
+			const notebookEditorDiffFactory: DiffEditorInputFactoryFunction = diffEditorInput => {
 				const modifiedInput = diffEditorInput.modifiedInput;
 				const originalInput = diffEditorInput.originalInput;
 				const notebookUri = modifiedInput.resource!;
 				const originalNotebookUri = originalInput.resource!;
-				return { editor: NotebookDiffEditorInput.create(this._instantiationService, notebookUri, modifiedInput.getName(), originalNotebookUri, originalInput.getName(), diffEditorInput.getName(), notebookProviderInfo.id) };
+				return { editor: NotebookDiffEditorInput.create(this._instantiationService, notebookUri, 'Modified Input Name', originalNotebookUri, 'Original Input Name', 'Diff Editor Input Name', notebookProviderInfo.id) };
```

Comment:
I think this can be solved by `NotebookDiffEditorInput` extending `DiffEditorInput` so that the name is properly computed. This seems just like an oversight that it isn't the case already.

Reply:
Yeah that's what my above comment said. I would have to ask Joh and Peng why they went with EditorInput.